org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/outline/ModelBuilder.scala

```diff
@@ -57,7 +57,7 @@ object ModelBuilder extends HasLogger {
         } else {
           sb.append("(")
           args.dropRight(1).foreach(x => { renderType(sb, x); sb.append(", ") })
-          sb.setLength(sb.length - 2)
+          sb.setLength(math.max(1, sb.length - 2))
```

Comment:
1 is the length of "(" and 2 is the length of ", ", which is added once too often by the previous line. I didn't know how to better fix the code without refactoring half of it.